sdk/cosmos/azure_data_cosmos/src/routing/global_endpoint_manager.rs

```diff
@@ -0,0 +1,614 @@
+    ///
+    /// # Summary
+    /// Evaluates whether the given request can be routed to multiple write regions based
+    /// on the request's operation type and resource type. Returns true only for write
+    /// operations on resources that support multi-master writes (documents and stored
+    /// procedure executions) when the account is configured for multiple write locations.
+    ///
+    /// # Arguments
+    /// * `request` - The Cosmos DB request to evaluate
+    ///
+    /// # Returns
+    /// `true` if the request can use multiple write locations, `false` otherwise
+    pub fn can_use_multiple_write_locations(&self, request: &CosmosRequest) -> bool {
+        !request.is_read_only_request()
+            && self
+                .can_support_multiple_write_locations(request.resource_type, request.operation_type)
+    }
+
+    /// Refreshes account properties and location information from the service.
+    ///
+    /// # Summary
+    /// Fetches the latest Cosmos DB account properties including regional endpoint information
+    /// and updates the location cache. Uses a Moka cache with 600 second TTL to avoid redundant
+    /// service calls. If `force_refresh` is true, invalidates the cache to ensure fresh data.
+    /// The location cache is updated only when new data is fetched (TTL expiry or forced refresh),
+    /// not when serving cached data.
+    ///
+    /// # Arguments
+    /// * `force_refresh` - If true, invalidates cache and forces fresh fetch from service
+    ///
+    /// # Returns
+    /// `Ok(())` if refresh succeeded, `Err` if fetching account properties failed
+    pub async fn refresh_location_async(&self, force_refresh: bool) -> Result<(), Error> {
+        // If force_refresh is true, invalidate the cache to ensure a fresh fetch
+        if force_refresh {
+            self.account_properties_cache
+                .invalidate(&ACCOUNT_PROPERTIES_KEY)
+                .await;
+        }
+
+        // When TTL expires or cache is invalidated, the async block executes and updates location cache
+        let _account_prop = self
+            .account_properties_cache
+            .try_get_with(ACCOUNT_PROPERTIES_KEY, async {
+                // Fetch latest account properties from service
+                let account_properties: AccountProperties =
+                    self.get_database_account().await?.into_body().json()?;
+
+                // Update location cache with the fetched account properties (only on fresh fetch)
+                {
+                    let mut cache = self.location_cache.lock().unwrap();
+                    cache.on_database_account_read(account_properties.clone());
+                }
+
+                Ok(account_properties)
+            })
+            .await
+            .map_err(|e: Arc<Error>| {
+                Error::with_message(
+                    azure_core::error::ErrorKind::Other,
+                    format!("Failed to fetch account properties: {}", e),
+                )
```

Comment:
I suggest using `Error::with_error` so we can provide the error value itself. It's fine to convert the `Arc<Error>` into an `Error` here by calling [`unwrap_or_clone`](https://doc.rust-lang.org/stable/std/sync/struct.Arc.html#method.unwrap_or_clone) since we're fairly likely to hold the only instance (making it free), and if we don't, we're in an error path and an extra clone isn't that concerning:


```suggestion
                Error::with_error(
                    azure_core::error::ErrorKind::Other,
                    e.unwrap_or_clone(),
                    format!("Failed to fetch account properties", e),
                )
```